app/src/main/java/org/mifos/mobilebanking/api/local/PreferencesHelper.java

```diff
@@ -29,14 +33,24 @@
     private static final String GCM_TOKEN = "gcm_token";
 
     private SharedPreferences sharedPreferences;
+    private Context context;
 
     @Inject
     public PreferencesHelper(@ApplicationContext Context context) {
+        this.context = context;
         this.sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
     }
 
     public void clear() {
-        sharedPreferences.edit().clear().apply();
+        SharedPreferences.Editor editor = sharedPreferences.edit();
+        //prevent deletion of url and tenant
+        for (Map.Entry<String, ?> entry: sharedPreferences.getAll().entrySet()) {
+            if (!(entry.getKey().equals(context.getString(R.string.pref_base_url_key)) ||
```

Comment:
Don't use key value name from strings.xml, please define key value on top of class like these https://github.com/openMF/mifos-mobile/pull/716/files#diff-4afc21547670976db853693ae9f5d17cR29